Improve and expand documentation of pipes

joshtriplett opened the pull request:
- Reference UNIX, not just Linux
- Simplify some of the language
- Don't imply that pipes *only* work across multiple processes; instead,
  *suggest* that they're typically used across two or more separate
  processes.
- Specify that portable applications cannot use multiple readers or
  multiple writers for messages larger than a byte, due to potential
  interleaving.

Tracking issue for anonymous pipes:
https://github.com/rust-lang/rust/issues/127154

joboet commented on library/std/src/pipe.rs:
```diff
@@ -1,18 +1,26 @@
 //!
-//! This module provides support for anonymous OS pipes, like [pipe] on Linux or [CreatePipe] on
-//! Windows.
+//! This module provides support for anonymous OS pipes, like [pipe] on UNIX/Linux or [CreatePipe]
```
I think we can consider Linux to be a UNIX variant for all intents and purposes here
```suggestion
//! This module provides support for anonymous OS pipes, like [pipe] on UNIX or [CreatePipe]
```

joshtriplett replied:
*shrug*, sure.

joboet reviewed (approved):
r=me with the comment nit applied

bors commented:
:umbrella: The latest upstream changes (presumably #135682) made this pull request unmergeable. Please [resolve the merge conflicts](https://rustc-dev-guide.rust-lang.org/git.html#rebasing-and-conflicts).

joshtriplett commented:
@bors r=joboet rollup

bors commented:
:pushpin: Commit ee77d7fff1b1ef264c3d954d1d82323b5e181adc has been approved by `joboet`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=joboet ee77d7fff1b1ef264c3d954d1d82323b5e181adc -->
<!-- homu: {"type":"Approved","sha":"ee77d7fff1b1ef264c3d954d1d82323b5e181adc","approver":"joboet","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

joshtriplett commented:
@bors r-

joshtriplett committed:
Improve and expand documentation of pipes

- Simplify some of the language
- Minor grammar fixes
- Don't imply that pipes *only* work across multiple processes; instead,
  *suggest* that they're typically used across two or more separate
  processes.
- Specify that portable applications cannot use multiple readers or
  multiple writers for messages larger than a byte, due to potential
  interleaving.
- Remove no-longer-referenced footnote URLs.

joshtriplett commented:
Rewrote to apply atop current HEAD, which had changed as a result of the move from `std::pipe` to `std::io`.

joshtriplett commented:
@bors r=joboet rollup

bors commented:
:pushpin: Commit 687607c31e7397b958ccf3c9a8699977b5b183fe has been approved by `joboet`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=joboet 687607c31e7397b958ccf3c9a8699977b5b183fe -->
<!-- homu: {"type":"Approved","sha":"687607c31e7397b958ccf3c9a8699977b5b183fe","approver":"joboet","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->
